src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Core/Badge/Badge.ts

```diff
@@ -262,11 +262,11 @@ export var badgeAssignment = (
                                         content_type: RIBadgeAssignment.content_type,
                                         data: {},
                                     };
-                                    assignment.data[SIBadgeAssignment.nick] = {
+                                    SIBadgeAssignment.set(assignment, {
```

Comment:
In both cases you get/set values on the `adhTopLevelState` service, not on a resource.